scripts/install-windows-prereqs.ps1

```diff
@@ -3,71 +3,109 @@
 
 Param(
```

Comment:
Could we please add a comment on how the hash values are generated so that someone else can update if needed? I am a little concerned on how maintainable is this.

Reply:
The hash values are calculated using Get-FileHash, which is called on each of the downloaded packages below to verify the known hashes for each package. 

I think I made an invalid assumption here that each of these downloaded files are tied to static URLs with specific versions. That isn't the case for some of these, like shellcheck and nuget.exe. I need to fix that!

For each downloaded file tied to a specific version (like Git-2.19.1-64-bit.exe), we should expect the contents of that file to not change. If a file like this, which has a specific version, does change from what we expect, then it is probable that the file has been changed maliciously.

As for the maintenance overhead of updating these packages to new versions, it's not much cost to calculate the SHA256 hash of a downloaded file alongside it. One could simply run: "curl -o myfile URL" followed by a "Get-FileHash myfile" to get the expected hash of that file.

I am strongly in favor of maintaining integrity checks of automated downloads of expected packages. Otherwise, someone who owns one of these third party endpoints could inject malicious code onto a customer's system without them knowing what was downloaded/installed. Moreover, some of these endpoints are not SSL/TLS, so a simple MITM attack is possible for those. Integrity checks are a must, in my opinion.

Reply:
I think the integrity checks make sense.  Maybe as a future improvement, as an idea, we could create a script that generates all these hash values to a version controlled yml file.  This yml file would be used as an input to both ansible file and this install script.  Then all we have to do is to update this new script.